rfc_code_review/comment_summary.md

```diff
@@ -0,0 +1,54 @@
+# RFC: Code Review Comments Summary
+
+## GitHub Missing Features
+
+* Intuitive Review Comments:
+  * While this feedback may be somewhat subjective and depend on personal preferences,
+    it was a common issue that was mentioned by many community members.  For example,
+    some comments disappear when a patch is updated, sometimes comments are auto-hidden
+    and can be easy to miss.  And it can be difficult to distinguish between general
+    review comments and comments on specific lines of code
+* Automatic Subscription Rules (Herald Rules)
+  * There is no builtin way for people to automatically subscribe to a pull request covering
+    a specific part of the codebase.  This is seen as a major downside of GitHub pull requests.
+* Configurable Notifications.
+  * Configuration for notifications is not very flexible and it can be difficult to manage especially
+     for contributors that subscribe to multiple projects.
+* Good support for "Stack Reviews".
+  * "Stack Reviews" are reviews of multiple dependant patches.  This is something that is easy to do
+     with Phabricator, but difficult to do with GitHub.
+
+## GitHub Positive Featues
+
+* Familiar for new contributors.
+  * Phabricator is seen by some as a barrier of entry for new contributors.
+* Can enforce developer policy / restrict pushes.
+  * With GitHub pull requests we can enforce parts of the developer policy with automated checks.
+* Third-party hosting and support
+  * Phabricator is no longer maintained upstream.
+* Easier to integrate with CI and other kinds of automation.
+  * GitHub actions makes it very easy to automate parts of our review process.
+  * One point that was made in the comments is that it is hard to have reliable pre-commit testing with Phabricator,
+    because Phabricator deals only with patches, so if the patch doesn't apply on the current main branch or some
+    other known commit then you can't run CI on the patch.  With GitHub pull requests everything is a branch, so
+    you can always run CI against them even if the patch is a little bit out-of-date with main.
+* Merge from UI
+  * You can merge pull requests directly from the web UI, which is much more simple than the Phabricator
+    process of manually (or with arc) downloading the patch, applying it and pushing.  This is both a time
+    saver and more intuitive for new contributors.
```

Comment:
If we have PRs and pre-merge testing, i would like to have an auto-submit option:
* The user sets a flag for auto-merge (e.g. via a comment or label)
* The PR is auto-merged, once some agreed conditions are met (e.g. pre-merge checks passing, at least n human approvals).

This would reduce the friction of pre-merge testing, especially for small patches.

Reply:
It's not clear in what you're saying that this helps new contributors who don't have commit access and so can't trigger the merge (neither through a button, and hopefully not with a comment!).
Also there a bunch of things up in the air in your answer: for example what will the commit message looks like if a PR is auto-merged?

> This would reduce the friction of pre-merge testing, especially for small patches.

I agree, I'm very supportive of a "fire-and-forget" submit queue being available. But that's not the claim here: there not mention of **testing** in this paragraph.

Reply:
>How is it "more intuitive for new contributors"? Since they can't merge the pull-request themselves anyway and will need someone to do it for them, it should be exactly identical?

This was mentioned in the comments on the issue.

Reply:
That does not seem enough of a reason to me to just phrase it this way.